applications/app/controllers/CampaignsController.scala

```diff
@@ -0,0 +1,47 @@
+package controllers
+
+import common._
+import conf.Configuration
+import model._
+import play.api.libs.json._
+import play.api.libs.ws.{WSClient, WSResponse}
+import play.api.mvc._
+
+import scala.concurrent._
+
+class CampaignsController(
+  val controllerComponents: ControllerComponents,
+  val wsClient: WSClient,
+)(implicit context: ApplicationContext)
+  extends BaseController with ImplicitControllerExecutionContext with Logging {
+
+  val root = Configuration.formstack.url + "/form/"
+  val endpoint = "/submission.json"
+  val token = Configuration.formstack.editorial.oauthToken
+
+  def formSubmit() = Action.async { implicit request: Request[AnyContent] =>
```

Comment:
If you wish, you can omit the type of `request`. The compiler should be able to infer it.